examples/gitops/argocd-multi-cluster/hub-cluster/main.tf

```diff
@@ -0,0 +1,372 @@
+
+  eks_cluster_id       = module.eks.cluster_name
+  eks_cluster_endpoint = module.eks.cluster_endpoint
+  eks_oidc_provider    = module.eks.oidc_provider
+  eks_cluster_version  = module.eks.cluster_version
+
+  enable_argocd         = true
+  argocd_manage_add_ons = true # Indicates addons to be install via ArgoCD
+  argocd_helm_config = {
+    namespace = local.argocd_namespace
+    version   = "5.23.1" # ArgoCD v2.6.2
+    values    = [local.argocd_values, local.argocd_sso]
+  }
+
+  # Add-ons
+  enable_aws_load_balancer_controller = true                                # ArgoCD UI depends on aws-loadbalancer-controller for Ingress
+  enable_metrics_server               = true                                # ArgoCD HPAs depend on metric-server
+  enable_external_dns                 = local.enable_ingress ? true : false # ArgoCD Server and UI use valid https domain name
+  external_dns_helm_config = {
+    domainFilters : [local.domain_name]
+  }
+  external_dns_route53_zone_arns = [local.argocd_domain_arn] # ArgoCD Server and UI domain name is registered in Route 53
+
+  # Observability for ArgoCD
+  enable_amazon_eks_aws_ebs_csi_driver = true
+  enable_prometheus                    = true
+
+  tags = local.tags
+}
+
+################################################################################
+# EKS Blueprints Add-Ons via ArgoCD
+################################################################################
+
+module "eks_blueprints_argocd_addons" {
+  source = "github.com/csantanapr/terraform-aws-eks-blueprints-addons//modules/argocd?ref=argo-multi-cluster" #TODO change git org to aws-ia
+
+  argocd_skip_install = true # Skip argocd controller install
+
+  helm_config = {
+    namespace        = local.argocd_namespace
+    create_namespace = false
+  }
+
+  applications = {
+    # This shows how to deploy Cluster addons using ArgoCD App of Apps pattern
+    addons = {
+      add_on_application = true
+      path               = "chart"
+      repo_url           = "https://github.com/csantanapr/eks-blueprints-add-ons.git"
+      target_revision    = "argo-multi-cluster" #TODO change main
+      #repo_url             = "git@github.com:csantanapr-test-gitops-1/eks-blueprints-add-ons.git" #TODO change to https://github.com/aws-samples/eks-blueprints-add-ons once git repo is updated
+      #ssh_key_secret_name  = "github-ssh-key" # Needed for private repos
+      #git_secret_namespace = "argocd"
+      #git_secret_name      = "${local.name}-addons"
+    }
+  }
+
+  addon_config = { for k, v in module.eks_blueprints_kubernetes_addons.argocd_addon_config : k => v if v != null }
+
+  addon_context = {
+    aws_region_name                = local.region
+    aws_caller_identity_account_id = data.aws_caller_identity.current.account_id
+    eks_cluster_id                 = module.eks.cluster_name
+  }
+
+  depends_on = [module.eks_blueprints_kubernetes_addons]
```

Comment:
this will be very problematic - each time `eks_blueprints_kubernetes_addons` changes in any way, this module will re-compute and any policies will be removed. we cannot use `depends_on` across modules  https://github.com/hashicorp/terraform/issues/30340#issuecomment-1010202582

Reply:
open to ideas how to solve the ordering for the argocd, addons, workloads